### PR TITLE
(maint) Standardize command-line options

### LINKED ISSUE #23: Should be consistent with Puppet CLI taxonomy
See puppet-nogui.

### PR DESCRIPTION
**Pull request by MikaelSmith**
Bring command-line options inline with UX guidelines.
Resolves #23.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
ping @nicklewis, mind taking a look and possibly merging?

**Comment by MikaelSmith** [comment]
Updated. @ahenroid interested in reviewing?

File: exe/driver.cc.in [hunk 1]
```diff
@@ -42,6 +42,21 @@ void help(po::options_description& desc)
...
+void conflicting_options(po::variables_map const& vm, string const& opt1, string const& opt2)
+{
+    if (vm.count(opt1) && !vm[opt1].defaulted() && vm.count(opt2) && !vm[opt2].defaulted()) {
+        throw po::error(opt1 + " and " + opt2 + " options conflict: please specify only one.");
+    }
+}
+
+template<typename... TOpts>
+void conflicting_options(po::variables_map const& vm, string const& opt1, string const& opt2, TOpts && ...opts)
+{
```
Comment (ahenroid) [inline exe/driver.cc.in:54]: The conflicting options check is a great idea. My only issue here is with the particulars of this
implementation. Do we really need the variadic template pattern here? Wouldn't it be just as well
to pass a list to a non-template function and iterate? I've certainly used this pattern myself, but I 
find it hard to parse for developers.

Comment (MikaelSmith) [reply exe/driver.cc.in:54]: Writing a full permutation comparison with a vector might actually be more complex. Want to propose an implementation?

Comment (ahenroid) [reply exe/driver.cc.in:54]: Well, I'd say stick with your implementation, but the snippet below I think accomplishes the same
goal, without pulling out the "big guns" of variadic function template with recursion. Also, your
implementation produces a much more specific error message, which is definitely good.

void conflicting_options(po::variables_map const& vm, vector<string>& opts)
{
  int count = 0;
  for (auto& opt : opts) {
    if (vm.count(opt) && !vm[opt].defaulted())
       count++;
  }
  if (count > 1)
     throw po::error(boost::format("options conflict: please specify only one of %1%.", boost::algorithm::join(opts, ", ")));
}

int
main()
{
    ...
   conflicting_options(vm, {"log-level", "quiet", "verbose", "debug"});
   ...
}

Comment (MikaelSmith) [reply exe/driver.cc.in:54]: The reduced complexity is nice. I think I'll stick with it for now for the better error message (and I don't think it's that tricky to understand). I wasn't thinking about the counting method, definitely got stuck on comparing every pair of options.
